File: projectBlueWater/src/main/java/com/lasthopesoftware/resources/strings/StringResourceProvider.java
package com.lasthopesoftware.resources.strings;

import android.content.Context;
import android.support.annotation.StringRes;

/**
 * Created by david on 7/3/16.
 */
public class StringResourceProvider implements IStringResourceProvider {
	private final Context context;

	public StringResourceProvider(Context context) {
		this.context = context;
	}

	@Override
	public String getString(@StringRes int stringResourceId) {
		return context.getString(stringResourceId);
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/resources/strings/StringResourceProvider.kt
package com.lasthopesoftware.resources.strings

import android.content.Context
import androidx.annotation.StringRes

/**
 * Created by david on 7/3/16.
 */
class StringResourceProvider(private val context: Context) : IStringResourceProvider {
    override fun getString(@StringRes stringResourceId: Int): String {
        return context.getString(stringResourceId)
    }
}
